src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -115,7 +157,14 @@ static ParquetBufferValue parseColumnValueToParquet(
         throw new SFException(
             ErrorCode.INVALID_FORMAT_ROW, type.getName(), "Passed null to non nullable field");
       }
-      stats.incCurrentNullCount();
+      if (type.isPrimitive()) {
+        if (!statsMap.containsKey(path)) {
+          throw new SFException(
+              ErrorCode.INTERNAL_ERROR,
+              String.format("Stats not found for column: %s", type.getName()));
+        }
+        statsMap.get(path).incCurrentNullCount();
```

Comment:
need the same incNullCount call for non-primitive fields too, no?

Reply:
The EPs from server and parquet footer does not record stats from non-primitive (non-leaf) fields.

Reply:
pl add a comment for this